touchforms/backend/xformserver.py

```diff
@@ -141,15 +139,25 @@ def do_GET(self):
     def cross_origin_header(self):
         if settings.ALLOW_CROSS_ORIGIN:
             self.send_header('Access-Control-Allow-Origin', '*')
-        
+
+
+def ensure_required_params(params, action, content):
+    for param in params:
+        if param not in content:
+            raise InvalidRequestException("'%s' is required for action: %s" % (param, action))
+
+
 def handle_request(content, server):
     if 'action' not in content:
-        return {'error': 'action required'}
+        ensure_required_params(['action'], 'All actions', content)
```

Comment:
woo! validating user input!
You can probably get rid of `if 'action' not in content:` here, right?